<a href="https://colab.research.google.com/github/Adwoa-22/lily/blob/main/my_first_ds.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
url = "https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv"
df = pd.read_csv(url)
df.head()

,total_bill,tip,sex,smoker,day,time,size
0,16.99,1.01,Female,No,Sun,Dinner,2
1,10.34,1.66,Male,No,Sun,Dinner,3
2,21.01,3.50,Male,No,Sun,Dinner,3
3,23.68,3.31,Male,No,Sun,Dinner,2
4,24.59,3.61,Female,No,Sun,Dinner,4


In [6]:
df.tail()

,total_bill,tip,sex,smoker,day,time,size
239,29.03,5.92,Male,No,Sat,Dinner,3
240,27.18,2.00,Female,Yes,Sat,Dinner,2
241,22.67,2.00,Male,Yes,Sat,Dinner,2
242,17.82,1.75,Male,No,Sat,Dinner,2
243,18.78,3.00,Female,No,Thur,Dinner,2


In [7]:
df.shape

(244, 7)

In [8]:
df.describe()

,total_bill,tip,size
count,244.000000,244.000000,244.000000
mean,19.785943,2.998279,2.569672
std,8.902412,1.383638,0.951100
min,3.070000,1.000000,1.000000
25%,13.347500,2.000000,2.000000
50%,17.795000,2.900000,2.000000
75%,24.127500,3.562500,3.000000
max,50.810000,10.000000,6.000000


In [10]:
df.groupby('time')['tip'].mean()

,tip
time,
Dinner,3.102670
Lunch,2.728088


In [11]:
df.groupby('time')['total_bill'].mean()

,total_bill
time,
Dinner,20.797159
Lunch,17.168676


In [12]:
df['tip_pct'] = df['tip'] / df['total_bill'] * 100


In [13]:
df.groupby('time')['tip_pct'].mean()

,tip_pct
time,
Dinner,15.951779
Lunch,16.412793


In [14]:
df.groupby('smoker')['tip_pct'].mean()

,tip_pct
smoker,
No,15.932846
Yes,16.319604


In [15]:
df.groupby('sex')['tip_pct'].mean()

,tip_pct
sex,
Female,16.649074
Male,15.765055


This dataset provides an overview of 244 rows and 7 column of restaurant vists. it looked at the total bill amount, the tip amount and othe factors that might have influenced tipping(time - whether the visit was lunch or dinner, smoking - whether the person smokes or not, the size of the of the party, and gender). After the analysis dinner tip was $3.10 while the lunch tip was $2.73 but upon applying percentage to it was seen that the lunch tip is slightly higher than the dinner tip.Gender showed a clear difference; female customers tipped about 16.6% on averag eversus 15.8% for male customers.  Smokers also tend to tip slightly higher than non smokers though this gap is small enough that it may not be a meaningful difference given the sample size